pkg/descheduler/strategies/node_affinity.go

```diff
@@ -33,6 +35,55 @@ import (
 
 const NodeAffinityName = "RemovePodsViolatingNodeAffinity"
 
+// NewRemovePodsViolatingNodeAffinity returns a StrategyController to run Node affinity with informers
+func NewRemovePodsViolatingNodeAffinity(
+	ctx context.Context,
+	client clientset.Interface,
+	sharedInformerFactory informers.SharedInformerFactory,
+	strategy api.DeschedulerStrategy,
+	podEvictor *evictions.PodEvictor,
+	nodeSelector string,
+	stopChannel chan struct{},
+) *StrategyController {
+	c := &StrategyController{
+		ctx:                   ctx,
+		client:                client,
+		queue:                 workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		sharedInformerFactory: sharedInformerFactory,
+		strategy:              strategy,
+		podEvictor:            podEvictor,
+		nodeSelector:          nodeSelector,
+		stopChannel:           stopChannel,
+	}
+	sharedInformerFactory.Core().V1().Nodes().Informer().AddEventHandler(nodeEventHandler(c))
+	return c
+}
+
+func nodeEventHandler(c *StrategyController) cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			nodes, err := nodeutil.ReadyNodes(c.ctx, c.client, c.sharedInformerFactory.Core().V1().Nodes(), c.nodeSelector)
+			if err != nil {
+				klog.V(1).InfoS("Unable to get ready nodes", "err", err)
+				close(c.stopChannel)
+				return
+			}
+			c.nodes = nodes
+			c.queue.Add(workQueueKey)
+		},
+		UpdateFunc: func(old, new interface{}) {
```

Comment:
Can you put down a comment saying `DeleteFunc is not needed since pods are automatically evicted` (or similar)